client/client_pool/src/concord_client_pool.cpp

```diff
@@ -146,8 +145,10 @@ SubmitResult ConcordClientPool::SendRequest(std::vector<uint8_t> &&request,
     }
   }
 
-  // Request hasn't been processed yet
-  if (external_requests_queue_.size() < jobs_queue_max_size_) {
+  // Request hasn't been processed yet and this is not a reconfiguration request then only allow the request to wait in
+  // the external request queue. Reconfigution requests are unbatched and in general require immediate response. So
+  // waiting is not acceptable.
+  if ((external_requests_queue_.size() < jobs_queue_max_size_) && (!(flags & ClientMsgFlag::RECONFIG_FLAG_REQ))) {
```

Comment:
Please talk to Lior regarding the external requests queue. AFAIR, it has to be removed.

Reply:
Thanks, @yuliasherman , yes i am discussing with @Liorgal28.